internal/core/metadata/rest_deviceservice_test.go

```diff
@@ -594,3 +691,20 @@ func createDeviceServiceRequestWithBody(
 
 	return mux.SetURLVars(req, pathParams)
 }
+
+func newMockDBClientFirst() interfaces.DBClient {
+	dbMock := &mocks.DBClient{}
+	dbMock.On("GetDeviceServiceById", testDeviceServiceLastUpdatedId).Return(testDeviceServiceLastUpdated, nil)
+	dbMock.On("GetDeviceServiceById", testDeviceServiceLastUpdatedNotExistId).Return(contract.DeviceService{}, testError)
+	dbMock.On("UpdateDeviceService", testDeviceServiceLastUpdated).Return(nil)
+
+	return dbMock
+}
+
+func newMockDBClientSecond() interfaces.DBClient {
+	dbMock := &mocks.DBClient{}
+	dbMock.On("GetDeviceServiceById", testDeviceServiceLastUpdatedId).Return(testDeviceServiceLastUpdated, nil)
+	dbMock.On("UpdateDeviceService", mock.Anything).Return(testError)
+
+	return dbMock
+}
```

Comment:
Can this be removed in favor of refactoring `createMockWithOutlines` to support mocking methods with varying number or return entities? It seems like we have outgrown the original function and it needs to be updated to support this. If we do want to keep these functions I think we should rename them better so it is clear what the difference is between them.

If we do refactor the `createMockWithOutlines` I think we can than 
 remove `TestUpdateServiceLastReportedByIdServiceUnavailable` in favor of a test table.

Reply:
I like the suggestion and will pursue the implementation 👍